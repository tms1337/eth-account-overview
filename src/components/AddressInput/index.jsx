import React from "react";
import { colors } from "../../config/ui";

const AddressInput = ({ title, onChange, registerField, isError }) => {
  const inputStyle = isError
    ? { ...styles.input, ...styles.errorInput }
    : styles.input;

  return (
    <div>
      <h2 style={styles.title}>{title}</h2>
      <input
        style={inputStyle}
        type="text"
        onChange={onChange}
        ref={registerField}
      />
    </div>
  );
};

const styles = {
  title: {
    fontSize: "25px",
    color: colors.PRIMARY
  },
  input: {
    width: "100%",
    height: "20px",
    appearance: "none",
    outline: "none",
    border: "none",
    borderBottom: "2px solid black",
    fontSize: "20px",
    color: colors.DARK_GREY
  },
  errorInput: {
    border: `2px solid ${colors.ERROR}`,
  }
};

export default AddressInput;
