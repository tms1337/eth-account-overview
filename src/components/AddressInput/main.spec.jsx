import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import AddressInput from ".";

describe("AddressInput", () => {
  it("renders snapshot", () => {
    const component = renderer.create(<AddressInput />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays input with title", () => {
    const title = "Enter your wallet address";
    const component = mount(<AddressInput title={title} />);

    expect(component.find("h2").length).toBe(1);
    expect(component.find("h2").text()).toBe(title);
    expect(component.find("input").length).toBe(1);
  });

  it("displays input with different title", () => {
    const title = "Different title";
    const component = mount(<AddressInput title={title} />);
    expect(component.find("h2").text()).toBe(title);
    expect(component.find("input").length).toBe(1);
  });

  it("calls the correct callback on change", () => {
    const title = "Holaho";
    const onChange = jest.fn();
    const component = mount(<AddressInput title={title} onChange={onChange} />);

    const input = component.find("input").at(0);

    const inputText = "0xaa";
    input.simulate("change", { target: { value: inputText } });

    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it("displays the submit button", () => {
    const component = mount(<AddressInput />);

    expect(component.find("input").length).toBe(1);
  });

  it("displays the submit button with correct title", () => {
    const submitTitle = "Go go!";
    const component = mount(<AddressInput submitTitle={submitTitle} />);

    expect(component.find("button").length).toBe(1);
    expect(
      component
        .find("button")
        .at(0)
        .text()
    ).toBe(submitTitle);
  });

  it("displays the submit button with correct another title", () => {
    const submitTitle = "Another Go!";
    const component = mount(<AddressInput submitTitle={submitTitle} />);

    expect(component.find("button").length).toBe(1);
    expect(
      component
        .find("button")
        .at(0)
        .text()
    ).toBe(submitTitle);
  });
});
