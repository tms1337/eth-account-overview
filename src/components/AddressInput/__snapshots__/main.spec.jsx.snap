// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInput renders snapshot 1`] = `
<div>
  <h2
    style={
      Object {
        "color": "#FF875B",
        "fontSize": "25px",
      }
    }
  />
  <input
    style={
      Object {
        "appearance": "none",
        "border": "none",
        "borderBottom": "1px solid black",
        "color": "#5c5b59",
        "fontSize": "20px",
        "height": "20px",
        "outline": "none",
        "width": "100%",
      }
    }
    type="text"
  />
</div>
`;
